.travis.yml

```diff
@@ -23,7 +23,30 @@ matrix:
     - python: 2.7
       env: TOXENV=docs
 
-install: pip install tox
+install:
+    - |
+        case "$TOXENV" in
+        "pypy")
+            PYPY_VERSION="pypy-2.5.0-linux64"
+            PYPY_BIN="pypy"
+            ;;
+        "pypy3")
+            PYPY_VERSION="pypy3.3-v5.2.0-alpha1-linux64"
+            PYPY_BIN="pypy3"
+            ;;
+        esac
+        if [ -n "$PYPY_VERSION" ] ; then
+            deactivate
+            pip install --upgrade --ignore-installed --user virtualenv
+            mkdir -p ~/pypy && pushd ~/pypy
+            wget "https://bitbucket.org/pypy/pypy/downloads/$PYPY_VERSION.tar.bz2"
+            tar jxf "$PYPY_VERSION.tar.bz2"
+            pushd "$PYPY_VERSION" && patch -i "$TRAVIS_BUILD_DIR/external/$TOXENV-subprocess.patch" -N -p0 -F0 && popd
+            virtualenv --python="$HOME/pypy/$PYPY_VERSION/bin/$PYPY_BIN" "$HOME/virtualenvs/$PYPY_VERSION"
```

Comment:
PyPy has patched ```subprocess.py```, which copies ```libpypy-c.so``` if pypy thinks the caller is virtualenv. To make the decision, pypy [checks ```sys.argv```](https://bitbucket.org/pypy/pypy/src/4afa4a9d8caffdd737e4b8f24c157a939c9127d2/lib-python/2.7/subprocess.py?at=default&fileviewer=file-view-default#subprocess.py-660). You can see virtualenv commit IDs (a9454bce, 0e9342ce, 10ba3f3c) in PyPy's source codes mentioned above. That's the truth - in some cases PyPy should be updated if virtualenv is changed. PyPy's architecture is silly, but I don't have many choices other than patching PyPy.